lib/analysis_options.yaml

```diff
@@ -252,18 +255,18 @@ linter:
     - unnecessary_string_escapes
     - unnecessary_string_interpolations
     - unnecessary_this
+    - unnecessary_to_list_in_spreads
     - unrelated_type_equality_checks
     - unsafe_html
     - use_build_context_synchronously
-    # linter v1.19+
     - use_colored_box
-    # linter v1.15+
     - use_decorated_box
     - use_full_hex_values_for_flutter_colors
     - use_function_type_syntax_for_parameters
     - use_if_null_to_convert_nulls_to_bools
     - use_is_even_rather_than_modulo
     - use_named_constants
+    - use_raw_strings
```

Comment:
This could potentially lead to annoyances like when you would want to add interpolation and it will not work. It feels like maybe it doesn't need to be enforced in the end.

Reply:
I'm not sure if it can lead to issues like that
e.g. this doesn't trigger the lint:
```dart
RegExp('${key}_(\\d+)u_1(m|y)\$')
```

I think it is safe to assume that it only triggers when escaped symbols aren't used unescaped in the same string

This will trigger:
```dart
RegExp('key_(\\d+)u_1(m|y)\$')
```

Reply:
what I mean if we force raw string and then would want to add interpolation (after some time).

Reply:
If we have a raw string somewhere:

```dart
final s = r'some complex text with different " 'quotes" ' and $ signs';
```

And then want to interpolate something there, we'll be forced to add escaping everywhere, instead of simply doing what we want:

```dart
final s = 'some complex ${someVar} text with different \" \'quotes\" \' and \$ signs';
```

This will make for non-clean diffs, and could be error-prone.

Reply:
Besides, even my example above shows some issues with quote handling in raw strings.

Reply:
This example doesn't trigger the lint, but I understand the issue with the diffs
```dart
'some complex text with different " \'quotes" \' and \$ signs';
```

Reply:
> This example doesn't trigger the lint

As in it won't force us to use a raw-string instead of a normal one here

Reply:
even this doesn't trigger a lint
```dart
'some complex text with different \'quotes \' and \$ signs';
```

(it can actually be a raw string):
```dart
r"some complex text with different 'quotes ' and $ signs";
```

Reply:
As for the diff issue - isn't the code read much more often than it is written?
IMO raw strings improve readability
Even if it is edited a few times after it was written, it will be probably read more times

WDYT?